new nim domain

connoralittle opened the pull request:
I have made a new nim domain, with both an efficient single pile nim and multi-pile domains

connoralittle committed:
new nim domain

haz commented:
I think ~30 problems per variant would be good (sub sample so we still have the same range of difficulty).

1. What's the difference between the different domains here?
2. Can you introduce the counter version of the problem (separate folder, I reckon, since both domain and problem would change)
3. You'll need to update the catalogue and README for it.

connoralittle committed:
new nim domain with counters

connoralittle commented:
0. I have removed structurally reduandant problems and have 30, 55, and 35 problems respectively for each heap.
1. Each domain corresponds to the number of heaps. Additional heaps include additional actions
2. I have added a counter version. It is much more efficient and structurally different so it may be worth scaling up the size of the problems irrespective of the original version.
3. I have added the updates.
